### PR TITLE
Fixes for php8

### PR DESCRIPTION
**Pull request by howdu**
Most notable change is in the Encryption class.

Amazed how little changes were required getting this to work on PHP8.1.

### CODE REVIEW

File: upload/catalog/model/catalog/product.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function getProduct($product_id) {
 				'height'           => $query->row['height'],
 				'length_class_id'  => $query->row['length_class_id'],
 				'subtract'         => $query->row['subtract'],
-				'rating'           => round($query->row['rating']),
+				'rating'           => $query->row['rating'] ? round($query->row['rating']) : 0,
```
Comment (arnisjuraga) [inline upload/catalog/model/catalog/product.php:50]: This should also work, isn't it?:

```  => round($query->row['rating'] ?? 0),```

File: upload/catalog/model/catalog/product.php [hunk 2]
```diff
@@ -594,4 +594,4 @@ public function getTotalProductSpecials() {
 		}
 	}
 }
-?>
\ No newline at end of file
+?>
```
Comment (arnisjuraga) [inline upload/catalog/model/catalog/product.php:597]: No need for closing tag at all. But hat would be much more global change in all files.

File: upload/catalog/model/catalog/product.php (hunk 1 above)
Comment (howdu) [reply upload/catalog/model/catalog/product.php:50]: Yes but some people may be still on PHP5

Comment (arnisjuraga) [reply upload/catalog/model/catalog/product.php:50]: Oh, OK. If this is meant to be also back compatible that will not work

**Comment by ADDCreative** [comment]
Encryption class has now been updated.

**Comment by howdu** [comment]
Cheers thanks for keeping this old version supported.

**Pull request closed without merging**